### ISSUE TITLE
[a11y]: Menu does not close on loss of focus. Potential WCAG 2.4.11 failure

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19300]
### Package

@carbon/react, @carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

n/a

### Description

Focus can be moved from within Menu to outside the component without the menu closing. If there is another element completely obscured by the menu that is brought into focus, this would fail WCAG 2.4.11. In order to prevent this failure, I believe either focus needs to be trapped within the menu until it is closed, or have the menu close automatically on loss of focus. Though, I'm not sure if this needs to be a change in our component or responsibility of the user to comply with WCAG.

https://github.com/user-attachments/assets/f0f964b2-5307-4a2c-bb83-0155b380b9e3

### WCAG 2.1 Violation

2.4.11 (Focus not obscured)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-menu--default

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-menu--default
2. Click somewhere on the menu (such as "Bold") to bring one of its options into focus
3. press TAB to move focus somewhere else on the page and notice that Menu doesn't close

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
